src/model-helpers/User.js

```diff
@@ -0,0 +1,173 @@
+import {
+  buildThing,
+  createThing,
+  setThing,
+  saveSolidDatasetAt,
+  getSolidDataset,
+  createSolidDataset,
+  getThingAll,
+  getDatetime,
+  getPodUrlAll,
+  getStringNoLocale,
+  getUrl
+} from '@inrupt/solid-client';
+import { RDF_PREDICATES } from '../constants';
+
+import { setDocAclForUser } from '../utils/network/session-helper';
+
+/**
+ * @typedef {import('@inrupt/solid-ui-react').SessionContext} Session
+ */
+
+/**
+ * Function that updates a user's last active time on Solid Pod
+ *
+ * @memberof User
+ * @function updateUserActivity
+ * @param {Session} session - Solid's Session Object {@link Session}
+ * @param {URL} podUrl - the url of the user's pod to update
+ * @returns {Promise} Promise - Updates last active time of user to lastActive.ttl
+ */
+export const updateUserActivity = async (session, podUrl) => {
+  const activityDocUrl = `${podUrl}public/active.ttl`;
+  const accessObject = {
+    read: true,
+    append: true,
+    write: true,
+    control: true
+  };
+  const publicAccessObject = {
+    read: true,
+    append: false,
+    write: false,
+    control: false
+  };
+  let activityDataset;
+
+  const updateAndSaveActivity = async () => {
+    const activity = buildThing(createThing({ name: 'active' }))
+      .addDatetime(RDF_PREDICATES.dateModified, new Date())
+      .build();
+
+    activityDataset = setThing(activityDataset, activity);
+
+    await saveSolidDatasetAt(activityDocUrl, activityDataset, {
+      fetch: session.fetch
+    });
+  };
+
+  try {
+    activityDataset = await getSolidDataset(activityDocUrl, { fetch: session.fetch });
+    await updateAndSaveActivity();
+  } catch {
+    activityDataset = createSolidDataset();
+    await updateAndSaveActivity();
+
+    await setDocAclForUser(
+      session,
+      activityDocUrl,
+      'create',
+      session.info.webId,
+      accessObject,
+      publicAccessObject
+    );
+  }
+};
+
+/**
+ * Fetches the last time a user was active in pass
+ *
+ * @memberof User
+ * @function loadUserActivity
+ * @param {URL} podUrl - Url of user's pod to fetch from
+ * @param {Session} session - Solid's Session Object {@link Session}
+ * @returns {string} last time the user was active in PASS
```

Comment:
Mentioned this during our discussion, JSDoc could take in types similar to TypeScript, so for Promises with string we can have `{Promise<string>}` in Line 84